### PR TITLE
Multirun Experiment UI

### PR DESCRIPTION
**Pull request by andrewmchen**
Implements a UI which nests runs who have a tag of `mlflow.parentRunID` which indicate their parent.
screen shot 2018-09-28 at 4 36 30 pm

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
This looks pretty good! Leaving some comments. Also had two high-level questions/thoughts:

* It'd be good for us to think about how this feature will work if we add pagination for runs within an experiment, although I think we could impose that requirement on the design for pagination (that it not break nesting parent/child runs)
* The animation when expanding/closing parent is really cool, but ends up lagging for me a bit when I generate a decent number of runs under a parent (currently testing with 30). I can show you offline

File: mlflow/server/js/src/components/ExperimentView.js [hunk 1]
```diff
@@ -6,23 +6,25 @@ import { getApis, getExperiment, getParams, getRunInfos, getRunTags } from '../r
...
 import { getLatestMetrics } from '../reducers/MetricReducer';
 import KeyFilter from '../utils/KeyFilter';
-import Utils from '../utils/Utils';
 
-import ExperimentRunsTableMultiColumn from "./ExperimentRunsTableMultiColumn";
-import ExperimentRunsTableCompact from "./ExperimentRunsTableCompact";
-import ExperimentViewUtil from "./ExperimentViewUtil";
-import ExperimentRunsSortToggle from "./ExperimentRunsSortToggle";
+import ExperimentRunsTableMultiColumnView from "./ExperimentRunsTableMultiColumnView";
+import ExperimentRunsTableCompactView from "./ExperimentRunsTableCompactView";
```
Comment (smurching) [inline mlflow/server/js/src/components/ExperimentView.js]: Small bug: I think we need to rename `ExperimentRunsTableCompact.js` -> `ExperimentRunsTableCompactView.js`

File: mlflow/server/js/src/components/ExperimentViewUtil.js [hunk 2]
```diff
@@ -98,8 +116,221 @@ export default class ExperimentViewUtil {
...
+    const runIdToIdx = {};
+    runInfos.forEach((r, idx) => {
+      runIdToIdx[r.run_uuid] = idx;
+    });
+    const treeNodes = runInfos.map(r => new TreeNode(r.run_uuid));
+    tagsList.forEach((tags, idx) => {
+      const parentRunId = tags['mlflow.parentRunId'];
+      if (parentRunId) {
+        const parentRunIdx = runIdToIdx[parentRunId.value];
+        if (parentRunIdx !== undefined) {
```
[the remark this answers is not in the document]
Comment (andrewmchen) [reply mlflow/server/js/src/components/ExperimentViewUtil.js]: This check is necessary for the case where the parent run is deleted. Then the child's pointer to the parent won't find anything.

File: mlflow/server/js/src/components/ExperimentViewUtil.js [hunk 3]
```diff
@@ -98,8 +116,221 @@ export default class ExperimentViewUtil {
...
+        return undefined;
+      }
+      current = current.parent;
+    }
+    return current;
+  }
+  isRoot() {
+    return this.findRoot().value === this.value;
+  }
+  isCycle() {
```
[the remark this answers is not in the document]
Comment (andrewmchen) [reply mlflow/server/js/src/components/ExperimentViewUtil.js]: Yeah, also if someone just decides to set the tag manually :)

**Review by smurching — APPROVED** [review]
LGTM! The UI data-generation script is great too :)

**Pull request merged**